Zend/zend_types.h

```diff
@@ -681,6 +681,7 @@ static zend_always_inline uint32_t zval_gc_info(uint32_t gc_type_info) {
 #define GC_IMMUTABLE                (1<<6) /* can't be changed in place */
 #define GC_PERSISTENT               (1<<7) /* allocated using malloc */
 #define GC_PERSISTENT_LOCAL         (1<<8) /* persistent, but thread-local */
+#define GC_DEBUG_PROTECTED          (1<<9) /* whether the protection is for debug functions */
```

Comment:
This makes conflict with ``IS_OBJ_FREE_CALLED``.

Reply:
Ah sorry I completely missed that it's reused by obj flags. I just had a quick try to do

```
diff --git a/Zend/zend_types.h b/Zend/zend_types.h
index 432c79f35a..9c4cf21ec1 100644
--- a/Zend/zend_types.h
+++ b/Zend/zend_types.h
@@ -636,9 +636,9 @@ static zend_always_inline zend_uchar zval_get_type(const zval* pz) {
 
 #define GC_TYPE_MASK                           0x0000000f
 #define GC_FLAGS_MASK                          0x000003f0
-#define GC_INFO_MASK                           0xfffffc00
+#define GC_INFO_MASK                           0xfffff800
 #define GC_FLAGS_SHIFT                         0
-#define GC_INFO_SHIFT                          10
+#define GC_INFO_SHIFT                          11
 
 static zend_always_inline zend_uchar zval_gc_type(uint32_t gc_type_info) {
        return (gc_type_info & GC_TYPE_MASK);
@@ -677,11 +677,11 @@ static zend_always_inline uint32_t zval_gc_info(uint32_t gc_type_info) {
 
 /* zval_gc_flags(zval.value->gc.u.type_info) (common flags) */
 #define GC_NOT_COLLECTABLE                     (1<<4)
-#define GC_PROTECTED                (1<<5) /* used for recursion detection */
-#define GC_IMMUTABLE                (1<<6) /* can't be changed in place */
-#define GC_PERSISTENT               (1<<7) /* allocated using malloc */
-#define GC_PERSISTENT_LOCAL         (1<<8) /* persistent, but thread-local */
-#define GC_DEBUG_PROTECTED          (1<<9) /* whether the protection is for debug functions */
+#define GC_PROTECTED                (1<<5)  /* used for recursion detection */
+#define GC_IMMUTABLE                (1<<6)  /* can't be changed in place */
+#define GC_PERSISTENT               (1<<7)  /* allocated using malloc */
+#define GC_PERSISTENT_LOCAL         (1<<8)  /* persistent, but thread-local */
+#define GC_DEBUG_PROTECTED          (1<<10) /* whether the protection is for debug functions */
 
 #define GC_NULL                                                (IS_NULL         | (GC_NOT_COLLECTABLE << GC_FLAGS_SHIFT))
 #define GC_STRING                                      (IS_STRING       | (GC_NOT_COLLECTABLE << GC_FLAGS_SHIFT))
```

It's likely missing some things as it's breaking something in GC because it shows leaks so I would need to investigate / debug it more but before I do that and spend more time on this, I wanted to check if you would be fine in general with extending common flags to 11 or you think there's a better solution for this? Basically I just don't want to waste time on something that you think is not the right solution.